### PR TITLE
Fixes date functions to be linux compatible

### PR DESCRIPTION
**Pull request by skyeagle**

### CODE REVIEW

File: lib/file.sh [hunk 1]
```diff
@@ -58,11 +58,11 @@ lib::file::install_with_backup() {
 }
 
 lib::file::last-modified-date() {
-  stat -f "%Sm" -t "%Y-%m-%d" "$1"
+  date -r $1 "+%Y-%m-%d"
```
Comment (jordanfbrown) [inline lib/file.sh:61]: This doesn't work on Mac

```
± % date -r $1 "+%Y-%m-%d"                                            
usage: date [-jnRu] [-d dst] [-r seconds] [-t west] [-v[+|-]val[ymwdHMS]] ...
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
```

Won't that cause issues in other places?

Comment (skyeagle) [reply lib/file.sh:61]: you have not passed filename as $1 argument

Comment (jordanfbrown) [reply lib/file.sh:61]: Doh 😞

**Review by jordanfbrown — APPROVED** [review]
LGTM!

**Pull request merged**